File: official/convex.workspace.runtime/implementations/convex_interlocking_infrastructure/fixtures/dirty/interlocking_no_cargo/convex/match/wagon.ts
// DIRTY fixture — coder.convex.interlocking-does-not-carry-cargo (wagon side).
// A wagon MUST NOT import interlocking code — that couples the artifact plane to route control.
// Expected: >=1 violation (wagon-imports-interlocking).
import { InterlockingRunner } from "../trains/interlocking";

export function runMatch(cargo: Map<string, unknown>): Map<string, unknown> {
  // BUG: wagon reaches into the route-control layer.
  const _runner = InterlockingRunner;
  cargo.set("match", { result: { validForRanked: true } });
  return cargo;
}

File: official/atdd.extension.coder.train-interlocking/implementations/interlocking-infrastructure/fixtures/dirty/python/match/wagon.py
"""match wagon — DIRTY fixture.

BUG: a wagon imports interlocking code, violating the Cargo boundary (core afokapu/atdd#1251).
"""
from trains.interlocking import InterlockingRunner  # forbidden: wagon imports interlocking


def run_match(cargo):
    _ = InterlockingRunner
    cargo["match"] = {"result": {"valid_for_ranked": True}}
    return cargo
